presto-main/src/test/java/com/facebook/presto/resourcemanager/TestResourceManagerClusterStateProvider.java

```diff
@@ -355,10 +445,12 @@ private static BasicQueryInfo createQueryInfo(String queryId, QueryState state,
 
     private static BasicQueryInfo createQueryInfo(String queryId, QueryState state, String resourceGroupId, MemoryPoolId memoryPool, DataSize totalMemoryReservation)
     {
+        String[] splits = resourceGroupId.split("\\.");
+        List<String> resourceGroupList = Arrays.asList(splits);
         return new BasicQueryInfo(
                 new QueryId(queryId),
                 TEST_SESSION.toSessionRepresentation(),
-                Optional.of(new ResourceGroupId(resourceGroupId)),
+                Optional.of(new ResourceGroupId(Arrays.asList(splits))),
```

Comment:
I would add a helper method on `ResourceGroupId` that removes the boilerplate of splitting and recreating an id.

Reply:
done as part of the constructor itself.